pkg/gatherers/common/gather_logs.go

```diff
@@ -17,25 +17,37 @@ import (
 	"github.com/openshift/insights-operator/pkg/utils/marshal"
 )
 
-// LogContainersFilter allows you to filter containers
-type LogContainersFilter struct {
-	Namespace                string
-	LabelSelector            string
-	FieldSelector            string
-	ContainerNameRegexFilter string
-	MaxNamespaceContainers   int
+// LogResourceFilter allows you to filter containers
+type LogResourceFilter struct {
+	// Deprecated: Namespace would be replaced by Namespaces in the future
+	Namespace                string   `json:"namespace"`
+	Namespaces               []string `json:"namespaces,omitempty"`
```

Comment:
It's not clear to me how this is supposed to work. Do you plan to iterate over the namespaces for the label selector value?
I think it will be essential to compile a list of messages for a given pod for a given namespace. I'm not sure how you're going to accomplish that. Can you please elaborate?

Reply:
> It's not clear to me how this is supposed to work. Do you plan to iterate over the namespaces for the pod name filter value?

Instead, one condition gathers logs from containers under one namespace it can get from multiple namespaces. This was one of the requests to reduce the need to create one condition per namespace.

> I think it will be essential to compile a list of messages for a given pod for a given namespace. I'm not sure how you're going to accomplish that. Can you please elaborate?

This is simple, I already implemented it on this PR:

```go
for _, namespace := range containersFilter.Namespaces {
```

It just iterates all the given Namespaces and fetches the logs. (yes, need probably some optimization as next step, but first we need to validate the whole implementation)

Reply:
Sorry but I can't see any benefit of this approach. Why should a rule developer provide a list of namespaces? I assume he/she should know the corresponding namespace for a particular pod.

Reply:
I can imagine two use cases where we would need to specify multiple namespaces:

* A platform pod has moved (or its namespace has been renamed) between OCP versions. This would be a fixed-length list of specific namespace names. IIUC, the current implementation supports this use case. On the other hand, we could solve this use case by splitting the gathering rule into multiple ones - one for each namespace. This case is quite rare so it shouldn't increase the config size too much.
* There is a grey zone of optional Red Hat-developed operators which customers can install with OLM into namespaces with custom names. In this case, we would need to search for the pod(s) in namespaces specified by other means then their names. Without some standardization on detecting such namespaces, these namespaces will be outside the hard-coded boundaries of what Insights Operator will be willing to gather. We don't need to worry about this use case right now, imho.

Having said that, how/where does the current implementation define the hard-coded boundaries?

Reply:
Currently, we are limited by `openshift-*` and `kube-*` namespaces.

> customers can install with OLM into namespaces with custom names

And based on this quote it seems that this boundary could be a problem... right?

Reply:
I would say let's limit the scope to 1 namespace per resource filter and `openshift-*` and `kube-*` namespaces for now.

We will need to deal with multiple rules for one pod anyway so we can solve the first use case by having multiple rules if we need it.

I wouldn't worry about the second use case right now. It is a separate issue (and probably a rather large one).